components/brave_new_tab_ui/containers/newTab/gridSites.tsx

```diff
@@ -42,12 +44,18 @@ const DynamicList = SortableContainer((props: DynamicListProps) => {
 
 class TopSitesList extends React.PureComponent<Props, {}> {
   onSortEnd = ({ oldIndex, newIndex }: SortEnd) => {
-    // do not update topsites order if the drag destination is a pinned tile
-    if (this.props.gridSites[newIndex].pinnedIndex) {
+    // User can't change order in "Most Visited" mode
+    if (!customLinksEnabled()) {
       return
     }
+    // Change the order in Chromium
+    reorderMostVisitedTile(this.props.gridSites[oldIndex].url, newIndex)
+    // Change the order that user sees. Chromium will overwrite this
+    // when `MostVisitedInfoChanged` is called- but changing BEFORE that
+    // avoids a flicker for the user where (for a second or so), tiles would
+    // have the wrong order.
```

Comment:
Consider moving this API call to the redux action handler for `tilesUpdated` so that this doesn't break for storybook (or we don't have to stub out when the topSites remote API isn't available)

Reply:
done!